### PR TITLE
feat: support lottie text

### PR DESCRIPTION
**Pull request by Dailoge**
support lottie text

### CODE REVIEW

**Comment by Dailoge** [comment]
已通过 example 福袋(金额文字) 验证

File: src/LottieResource.ts [hunk 1]
```diff
@@ -45,6 +47,24 @@ export type TypeRes = {
   lolitaAnimations?: TypeAnimationClip[];
 };
 
+export type TypeText = {
+  d: {
+    k: TypeTextKeyframe[];
+  }
+}
+
+// 字段解释：https://lottiefiles.github.io/lottie-docs/text/#text-document
```
Comment (eyworldwide) [inline src/LottieResource.ts:56]: 注释改成英文

File: src/element/TextLottieElement.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if (entity) {
+      this.entity = entity;
+      if (name) {
+        this.entity.name = name;
+      }
+    } else {
+      this.entity = new Entity(engine, layer.nm);
+    }
+    const textRenderer = this.entity.addComponent(TextRenderer);
+    const firstKeyframe: TypeTextKeyframe = layer?.t?.d?.k?.[0];
```
Comment (eyworldwide) [inline src/element/TextLottieElement.ts:21]: You should add a condition before this line to make it  more compatible:
```typescript
if(layer?.t?.d?.k?.length === 1) {
    // only one frame
}
else {
    // TODO: multi keyframes
}
```

File: src/element/TextLottieElement.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const firstKeyframeStart = firstKeyframe.s;
+    if (firstKeyframeStart) {
+      const { t: text, f: font, s: fontSize, fc: fontColor, lh: lineHeight } = firstKeyframeStart;
+      // 通过 font 设置 Font 对象
```
Comment (eyworldwide) [inline src/element/TextLottieElement.ts:25]: Use english too.

File: src/element/TextLottieElement.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      textRenderer.lineSpacing = lineHeight;
+    } else {
+      Logger.warn(`TextLottieElement: ${name} 未找到对应文字数据`);
+    }
+
+    textRenderer.priority = (Number.MAX_SAFE_INTEGER - this.index * 1000000) / Number.MAX_SAFE_INTEGER;
+  }
+
+  destroy() {
+    super.destroy();
```
Comment (eyworldwide) [inline src/element/TextLottieElement.ts:43]: You can delete the function if you have no additional code to execute.

**Comment by eyworldwide** [comment]
update the  **version** in package.json when you have resolved all conversations.

File: src/element/TextLottieElement.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        textRenderer.fontSize = fontSize;
+        // set the text color by color
+        textRenderer.color.set(fontColor[0], fontColor[1], fontColor[2], 1);
+        // set line spacing via lineSpacing
+        textRenderer.lineSpacing = lineHeight;
+      } else {
+        Logger.warn(`TextLottieElement: ${name}, No corresponding text data found`);
+      }
+    } else {
+      // TODO: multi keyframes
```
Comment (eyworldwide) [inline src/element/TextLottieElement.ts:41]: Add log to tell this feature is not supported in this version.

**Review by singlecoder — APPROVED** [review]
+1

**Comment by singlecoder** [comment]
When set priority, should update all renderers's priority in LottieAnimation onUpdate function

**Comment by eyworldwide** [comment]
> When set priority, should update all renderers's priority in LottieAnimation onUpdate function

For now there is no need to do this because `index` filed is hard code in lottie JSON.

**Review by eyworldwide — APPROVED** [review]

**Comment by eyworldwide** [comment]
@galacean/engine-lottie@1.0.0-beta.13 published

**Pull request merged**